### DISCUSSION TITLE
Testing / porting plans?

### DISCUSSION THREAD

**Discussion by znmeb** [pschatzmann/Adafruit_TinyUSB_Arduino#1] [Q&A] [answered]
I have a few RP2350 boards I can test USB audio on. I have a Pimoroni Pico Plus 2 and Plus 2 W, a SparkFun Pro Micro RP2350, an Adafruit Feather RP2350 with HSTX, and the two iLabs Challenger+ RP2350 boards.

I'll probably test on the Feather first. Is there a test / example sketch I can use?

**Comment by pschatzmann** [comment, marked as answer]
Delete the Adafruit_TinyUSB_Arduino folder in the packages/rp3040/4.0.3/libraries. Please note that ths is different than the regular libraries directoy into which you install libraries normally. TinyUSB is part of the core and must be replaced there.
In my linux this is e.g.

    sudo rm -r ~/.arduino15/packages/rp2040/hardware/rp2040/4.1.1/libraries/Adafruit_TinyUSB_Arduino
    cd ~/.arduino15/packages/rp2040/hardware/rp2040/4.1.1/libraries
 
In the libraries directory of the arduino packages execut the following commands

    git clone https://github.com/pschatzmann/Adafruit_TinyUSB_Arduino
    cd Adafruit_TinyUSB_Arduino
    git switch Audio

The Audio microphone examples should work!
You can easily extend the existing tests with the functionality from the [AudioTools](https://github.com/pschatzmann/arduino-audio-tools),
 
You get back to the original main branch with

    git switch master
 
ps. So far I have enabled this only for the RP2040 Architecture: this should conver the RP2350 as well.

**Reply by znmeb** [reply]
OK - sounds good. I've tested the 4.0.x and 4.1.1 arduino-pico core with a few RP2350 boards from PlatformIO with no problems on the Arduino side, and the Arduino RP2040 USB serial test sketches work on RP2350 boards so I'm guessing this will work.

**Comment by pschatzmann** [comment]
Not sure what PlatformIO will do here: just make sure that you specify the correct branch.
I was testing with Arduino recording audio with Audacity on my Linux Laptop

**Reply by pschatzmann** [reply]
The speaker example is working now as well and the issues for the non working cdc Serial output has been corrected

**Comment by znmeb** [comment]
Good to hear. I'm rebuilding my development container so I can pull in the new GitHub repo.

**Comment by pschatzmann** [comment]
Do you have any news ?

From my side: I finally managed to have the Headset scenario working together with CDC as well. It took me quite some time to figure out  that the default desciptor buffer was too small and that I needed to define my own buffer.

Today I did a quick test with a Seeeduino Xiao: and there it is working w/o problems as well...

**Comment by znmeb** [comment]
First test: I had a few missing pieces, most notably figuring out how to tell `arduino-cli` which USB stack to use, but I've got the test setup working. For the record, you tack it onto the end of the FQBN - `rp2040:rp2040:adafruit_feather_rp2350_hstx:usbstack=tinyusb`

I went straight for the `audio_microphone_stream` example since it's closest to my eventual use case - a synthesizer that plugs into a USB port. The compiler was not happy ... see attached logfile.

I'm going to try the one without the sawtooth - `audio_microphone` - to see if I can get a clean compile, upload and white noise coming into the laptop. I'm currently using an Adafruit Feather RP2350 HSTX as a test board but I have other boards with attached I2S audio adapters or Bluetooth that I could use to test Arduino Audio Tools.

[audio_microphone_stream.zip](https://github.com/user-attachments/files/17441650/audio_microphone_stream.zip)

**Reply by pschatzmann** [reply]
I committed the SawToothGenerator today, so you might need to upgrade the AudioTools, or replace SawToothGenerator with a SineWaveGenerator whch should exist in your installed version.

**Reply by znmeb** [reply]
Ah ... turns out I was installing from the release tag rather than from the main branch! It records and plays back in Audacity now.

**Comment by znmeb** [comment]
Good news - `audio_microphone` is working! It compiles, uploads, writes its log to the serial port, shows up as a microphone in Audacity and is clearly delivering white noise to the recorded track.

**Comment by znmeb** [comment]
I also have some boards supported by [espressif/arduino-esp32](https://github.com/espressif/arduino-esp32). Is there any easy way I can test with one of them?

**Reply by pschatzmann** [reply]
I did some testing myself and did not manage to get it working and it would need to be a S2 or S3
I think I need to build my custom Arduino core to make this work there.

It is working well on all the Arduino cores that have the USB selection in the Arduino Tools menu.

**Comment by znmeb** [comment]
Ah, OK. I have an M5Stack Cardputer (ESP32-S3) and an ESP32-C6 DevKit C-1. But I'm in no hurry for it, unless you have boards to test on yourself.

**Comment by pschatzmann** [comment]
Cool, that increases the list of tested working microcontrollers to
- RP2040
- RP2350
- SAMD21 (Seeeduino XIAO)

So the next challang would be to get to grips with the ESP32 Library Builder or maybe use Arduino as a Component

**Comment by znmeb** [comment]
I also have a Teensy 4.1. It has its own extensive audio library and I think it has USB audio, but nothing like the Synthesis Tool Kit.

But the really interesting one is the [M5Stack Cardputer](https://shop.m5stack.com/products/m5stack-cardputer-kit-w-m5stamps3). It has a tiny speaker, a microphone, WiFi and Bluetooth, a tiny display, an infrared transmitter and an SD card slot!
